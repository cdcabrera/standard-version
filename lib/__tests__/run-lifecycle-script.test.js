const runLifecycleScriptTest = require('../run-lifecycle-script');

describe('run-lifecycle-script', () => {
  it('should attempt to exec commands', async () => {
    const mockInfo = jest.fn();
    const spyInfo = jest.spyOn(console, 'info').mockImplementation(mockInfo);

    const mockWarn = jest.fn();
    const spyWarn = jest.spyOn(console, 'warn').mockImplementation(mockWarn);

    const scriptMismatchResults = await runLifecycleScriptTest({ scripts: { lorem: 'echo "mismatch"' } }, 'dolor');
    const dryRunResults = await runLifecycleScriptTest({ dryRun: true, scripts: { lorem: 'echo "dry run"' } }, 'lorem');
    const echoCmdResults = await runLifecycleScriptTest({ scripts: { lorem: 'echo "lorem ipsum"' } }, 'lorem');
    const errorResults = await runLifecycleScriptTest(
      { scripts: { lorem: 'echo "dolor sit error" && exit 0' } },
      'lorem'
    );
    const errorResultsStdOut = await runLifecycleScriptTest(
      { scripts: { lorem: 'echo "dolor sit stdout error" 1>&2 && exit 0' } },
      'lorem'
    );

    expect({
      scriptMismatchResults,
      dryRunResults,
      echoCmdResults,
      errorResults,
      errorResultsStdOut
    }).toMatchSnapshot('command results');

    expect({ info: mockInfo.mock.calls, warn: mockWarn.mock.calls }).toMatchSnapshot('console messages');

    spyInfo.mockClear();
    spyWarn.mockClear();
  });
});
