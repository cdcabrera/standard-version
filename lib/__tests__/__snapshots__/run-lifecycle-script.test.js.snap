// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run-lifecycle-script should attempt to exec commands: command results 1`] = `
Object {
  "dryRunResults": undefined,
  "echoCmdResults": "lorem ipsum
",
  "errorResults": "dolor sit error
",
  "errorResultsStdOut": "",
  "scriptMismatchResults": undefined,
}
`;

exports[`run-lifecycle-script should attempt to exec commands: console messages 1`] = `
Object {
  "info": Array [
    Array [
      "<green>✔</green> Running lifecycle script \\"<bold>lorem</bold>\\"",
    ],
    Array [
      "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"dry run\\"</bold>\\"",
    ],
    Array [
      "<green>✔</green> Running lifecycle script \\"<bold>lorem</bold>\\"",
    ],
    Array [
      "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"lorem ipsum\\"</bold>\\"",
    ],
    Array [
      "<green>✔</green> Running lifecycle script \\"<bold>lorem</bold>\\"",
    ],
    Array [
      "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"dolor sit error\\" && exit 0</bold>\\"",
    ],
    Array [
      "<green>✔</green> Running lifecycle script \\"<bold>lorem</bold>\\"",
    ],
    Array [
      "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"dolor sit stdout error\\" 1>&2 && exit 0</bold>\\"",
    ],
  ],
  "warn": Array [
    Array [
      "<yellow>dolor sit stdout error
</yellow>",
    ],
  ],
}
`;
